redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/upgrade/event/RedbeamsStartUpgradeRequest.java

```diff
@@ -0,0 +1,25 @@
+package com.sequenceiq.redbeams.flow.redbeams.upgrade.event;
+
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsEvent;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.RedbeamsUpgradeEvent;
+
+public class RedbeamsStartUpgradeRequest extends RedbeamsEvent {
+
+    private final String majorVersion;
```

Comment:
I know this is the previous PR, but what is the advantage of having `majorVersion` as String over an Enum value? 
I'd go with an enum as it is _very_ slowly changing and validation for the String content is missing, too..

Reply:
Would you have the enum already on the REST endpoint?

Reply:
Yes, that would be ideal and fool-proof, what's your take?